include/eeprom/core/task.hpp

```diff
@@ -85,10 +177,9 @@ class EEPromTask {
     /**
      * Task entry point.
      */
-    template <message_writer_task::TaskClient CanClient>
     [[noreturn]] void operator()(i2c::writer::Writer<QueueImpl> *writer,
-                                 CanClient *can_client) {
-        auto handler = EEPromMessageHandler{*writer, *can_client, get_queue()};
+                                 write_protect::WriteProtectPin *pin) {
+        auto handler = EEPromMessageHandler{*writer, get_queue(), *pin};
```

Comment:
I'm very confused as to which queue is receiving the eeprom can messages. Don't we still need the queue client so we can actually receive messages from CAN?

Reply:
got it